libsolidity/analysis/TypeChecker.cpp

```diff
@@ -299,7 +299,7 @@ void TypeChecker::checkContractIllegalOverrides(ContractDefinition const& _contr
 					overriding->isPayable() != function->isPayable() ||
 					overridingType != functionType
```

Comment:
Indeed, it's a violation of DRY. We can turn this into a `checkFunctionOverride(overriding, function)` that would trigger type errors conditionally.